python/openinference-instrumentation/src/openinference/instrumentation/config.py

```diff
@@ -343,10 +353,32 @@ def end(self, end_time: Optional[int] = None) -> None:
         span.end(end_time)
 
 
+class _IdGenerator(IdGenerator):
+    """
+    An IdGenerator that uses a different source of randomness to
+    avoid being affected by seeds set by user application.
+    """
+
+    def generate_span_id(self) -> int:
+        while (span_id := randbits(64)) == INVALID_SPAN_ID:
+            continue
```

Comment:
what exactly is an invalid_span_id?

Reply:
it's all [zeros](https://github.com/open-telemetry/opentelemetry-python/blob/9697f805d1e5ba53cc55e2502ad1de0cc7c5900a/opentelemetry-api/src/opentelemetry/trace/span.py#L578-L579)

looping is based on otel [implementation](https://github.com/open-telemetry/opentelemetry-python/blob/9697f805d1e5ba53cc55e2502ad1de0cc7c5900a/opentelemetry-sdk/src/opentelemetry/sdk/trace/id_generator.py#L52)

Reply:
Got it, thanks for the link